Updated NGINX OSS installation procedure. (#568)
nginx/documentation: content/nginx/admin-guide/installing-nginx/installing-nginx-open-source.md

- how-to
---

> This article explains how to install NGINX Open Source on various operating systems, including an overview of existing NGINX Open Source versions, installation types and methods, modules included in the default package and dynamic modules packages, and the basics of compiling nginx from the source code.

Reviewer: ```suggestion
This article explains how to install NGINX Open Source on various operating systems, including an overview of existing NGINX Open Source versions, installation types and methods, modules included in the default package and dynamic modules packages, and the basics of compiling NGINX from the source code.
```

Reviewer: Later in the file, I see you've used an alternative that IMO is just as good: 

```
`nginx`
```

## Stable and mainline versions {#stable_vs_mainline}

> NGINX Open Source is available in two versions: Mainline and Stable. The source code and release notes for both versions are available from [nginx download page](https://nginx.org/en/download.html).

Reviewer: ```suggestion
NGINX Open Source is available in two versions: Mainline and Stable. The source code and release notes for both versions are available from [NGINX download page](https://nginx.org/en/download.html).
```

### Mainline version

> Mainline version (also Mainline release, Mainline branch) is the latest development version, updated approximately every 1 or 2 months, includes the latest features, bug fixes and security fixes. This version is recommended for production unless your organization has strict requirements for stability, in which case the stable version is the better choice. The Mainline version is always odd-numbered, for example, 1.*27*.X.

Reviewer: Nitpick: Even though I see what you've bolded, I see "odd-number", and I think, "Oh, OK, the first 1 is an odd number." I wonder if you can describe it differently, something like:

```suggestion
Mainline version (also Mainline release, Mainline branch) is the latest development version, updated approximately every 1 or 2 months, includes the latest features, bug fixes and security fixes. This version is recommended for production unless your organization has strict requirements for stability, in which case the stable version is the better choice. The Mainline version always has an odd middle number, for example, 1.*27*.X.
```

I can visualize similar options such as "odd minor version number" or "odd second number"

##  Installation methods {#compile_vs_package}

Both the NGINX Open Source Mainline and Stable versions can be installed in several ways:

- A package from the [official NGINX Open Source repository](#install-from-the-official-repository) (recommended for production). This is the most reliable method: you have to set up the repository once, but after that the provided package is always up to date.

- A package from your [operating system’s default package repository](#install-from-the-default-system-repository) (suitable for learning or testing). This is the easiest method, but generally the provided package is outdated.

- Your own package [compiled from source](#sources) (recommended for advanced and custom builds). This method is the most flexible: you can add non-standard modules, including third‑party modules, or apply the latest security patches.

> ## Package contetns {#prebuilt}

Reviewer: ```suggestion
## Package contents {#prebuilt}
```

| [`ngx_http_perl_module`](https://nginx.org/en/docs/http/ngx_http_perl_module.html) | Implements location and variable handlers in Perl and inserts Perl calls into SSI. | `nginx-module-perl` |
| [`ngx_http_xsl_module`](https://nginx.org/en/docs/http/ngx_http_xslt_module.html) | Transforms XML responses using one or more XSLT stylesheets. | `nginx-module-xslt` |
| [`ngx_otel_module`](https://nginx.org/en/docs/ngx_otel_module.html) | Provides OpenTelemetry distributed tracing support. | `nginx-module-otel` |
{{</bootstrap-table>}}

## Install from the default system repository

Installing from your operating system’s default repository is the easiest method, but the package it provides is often outdated. To compare release versions and features, see [Changelog](https://nginx.org/en/CHANGES). For the most current version, it is recommended to install from the official NGINX repository instead.

> The steps include updating the package repository and installing nginx.

Reviewer: ```suggestion
The steps include updating the package repository and installing NGINX.
```

> - For MacOS:

Reviewer: Question: does this suggest that we support MacOS?

Author: it's up to operating systems / package managers to maintain their repos. The reason why it's here is:

- it's the easiest way to get nginx for testing, learning, demoing, etc. Now, each laptop is either mac or windows - if someone wants to quickly explore nginx, there's no need for vms, linux machine, or compiling from source

- I use nginx from brew for testing regularly, and it's good and always up-to-date, compared to other distros

- the reason why installing from package managers is here is
  1) it's a popular way of installing
  2) the package there is generally outdated, so for production we recommend our repo
  3) there may be questions why we need your repo while it's available from package managers
  4) nginx can be compiled/build/installed on almost any unix system (you just need to compile it yourself or install from pkg manager), while we provide up-to-date packages for most popular production oses
